frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -629,21 +627,30 @@ export default ({
       this.messages.splice(index, 1)
       this.handleSendMessage(message.text, message.attachments, message.replyingMessage)
     },
-    replyMessage (message) {
-      const { text, hash, type } = message
+    replyToMessage (message) {
+      const { text, hash, type, attachments } = message
+      const isTypeInteractive = type === MESSAGE_TYPES.INTERACTIVE
 
-      if (type === MESSAGE_TYPES.INTERACTIVE) {
+      if (isTypeInteractive) {
         const proposal = message.proposal
 
         this.ephemeral.replyingMessage = {
-          text: interactiveMessage(proposal, { from: `${CHATROOM_MEMBER_MENTION_SPECIAL_CHAR}${proposal.creatorID}` }),
-          hash
+          hash, text: interactiveMessage(proposal, { from: `${CHATROOM_MEMBER_MENTION_SPECIAL_CHAR}${proposal.creatorID}` })
         }
-        this.ephemeral.replyingTo = L('Proposal notification')
+      } else if (!text && attachments.length) {
```

Comment:
The message contracts define `attachments` as an `optional`, so could accessing `attachments.length` could result in a bug?

If so, then either here we need `attachments?.length` or above a default value:

```js
const { text, hash, type, attachments = [] } = message
```

Reply:
> It's impossible that a message has no text content but does not have attachments either

If you want to make this assumption, I strongly suggest that you either add this to the chatroom contract validation, or else that you don't make the assumption.

Reply:
@corrideat Sounds a bit rude man... what's your intention of randomly popping up and saying that..?

Reply:
Sorry, I missed the second part of the message where you said you'll make the change anyhow, so I can see why it could come across like that. I was looking through open PRs and saw the part about the send button not being enabled and somehow missed the second part. Anyhow, the motivation was that the chatroom contract will be used by other applications possibly, and for robustness it should not assume things that aren't part of the contract.

Reply:
@corrideat No worries.

> the motivation was that the chatroom contract will be used by other applications possibly, and for robustness it should not assume things that aren't part of the contract.

This sounds like an important point too.

Reply:
That'll teach me to make sure I read everything in full. BTW, another situation where this situation (attachments but no message body) might arise is https://github.com/okTurtles/group-income/issues/1900. However, we might need to treat that especially to work with this PR anyhow.

Reply:
@corrideat What I would love to mention here though is how things can sound. I've actually felt this a few times before that when you wrote `Hey I think this is wrong and you should do this` kinds of comments, they felt kind of disrespecful for some reason(especially when apparently there were no request for you to review the PR, even though I usually appreciate other dev's feedbacks). Maybe it's just that you just tend to say things bluntly and I've been more used to co-worker's style trying to be more polite than usual when they want to say someone else is wrong and they are right. But I can see you have no intention to harm other person's feeling so, no worries bro!